src/components/Modal.astro

```diff
@@ -0,0 +1,90 @@
+---
+interface Props {
+  modalContent: {
+    title: string;
+    description: string;
+  };
+}
```

Comment:
Title and description should be separate props, in the template a slot should take precedence